src/lib/mathlib/math/Matrix.hpp

```diff
@@ -491,6 +491,49 @@ class __EXPORT Matrix<3, 3> : public MatrixBase<3, 3>
 
 		return euler;
 	}
+
+	/**
+	 * get euler angles from rotation matrix and avoid gimbal lock effects
+	 * euler.data[*] are the elements of the rotation matrix from body to nav frame (NED), i.e. Xn = Tbn * Xb
+	 * This routine checks for the Aircrafts flight regime and rotates the coordinate frame accordingly to avoid gimbal lock effects.
+	 * In plots you will notice a sudden 90° pitch angle change and roll/yaw axis of the body frame switch. Roll/Yaw in the NED framne remain the same.
+	 * coordinate_frame 0 = nose down
```

Comment:
I presume these coordinate frames mean that they are a rigid body rotation, or are you just changing the euler extraction logic? Or in other words, are the euler angles this exports still unique, or is there a zero-degree pitch for hover and a zero degree pitch for level flight, but the nose is in one case up and in one case front?